### PR TITLE
Using upstream hostname for request (#294)

### PR DESCRIPTION
**Pull request by mitchelldavis**

### CODE REVIEW

**Comment by mitchelldavis** [comment]
* So if the two tests are redundant, should I roll them into one or just get rid of the second?
* I set UseUpstreamHostname on line 73 in `route.go`.  Is that not correct?
* `usehost` works for me and the only one I can come up with otherwise is `resetHost`.

**Comment by magiconair** [comment]
You can replace the tests with this:

```go
func TestProxyUsehost(t *testing.T) {
	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		fmt.Fprint(w, r.Host)
	}))

	proxy := httptest.NewServer(&HTTPProxy{
		Transport: &http.Transport{
			Dial: func(network, addr string) (net.Conn, error) {
				addr = server.URL[len("http://"):]
				return net.Dial(network, addr)
			},
		},
		Lookup: func(r *http.Request) *route.Target {
			routes := "route add mock /usehost http://a.com/ opts \"usehost\"\n"
			routes += "route add mock / http://a.com/"
			tbl, _ := route.NewTable(routes)
			return tbl.Lookup(r, "", route.Picker["rr"], route.Matcher["prefix"])
		},
	})
	defer proxy.Close()

	check := func(uri, host string) {
		resp, body := mustGet(proxy.URL + uri)
		if got, want := resp.StatusCode, http.StatusOK; got != want {
			t.Fatalf("got status %d want %d", got, want)
		}
		if got, want := string(body), host; got != want {
			t.Fatalf("got body %q want %q", got, want)
		}
	}

	proxyHost := proxy.URL[len("http://"):]
	t.Run("use host", func(t *testing.T) { check("/usehost", "a.com") })
	t.Run("no host", func(t *testing.T) { check("/", proxyHost) })
}
```

2. I mean to use `usehost` instead of `usehost=(true|false)`. This changes on how you set the `t.UseHost` flag since checking for an empty string is not sufficient. 

3. `usehost` it is then. Pls also use `t.UseHost` internally.

Thx a lot

File: route/route.go [hunk 1]
```diff
@@ -71,7 +71,11 @@ func (r *Route) addTarget(service string, targetURL *url.URL, fixedWeight float6
 	if r.Opts != nil {
 		t.StripPath = r.Opts["strip"]
 		t.TLSSkipVerify = r.Opts["tlsskipverify"] == "true"
-		t.UseUpstreamHostname = r.Opts["useupstreamhostname"] == "true"
+		// If the 'usehost' key exists in the map, we set
```
Comment (magiconair) [inline route/route.go]: This is redundant. Just use what I wrote initially

```go
_, t.UseHost = r.Opts["usehost"]
```

File: route/target.go [hunk 2]
```diff
@@ -21,10 +21,10 @@ type Target struct {
 	// TLS connections.
 	TLSSkipVerify bool
 
-	// UseUpstreamHostname causes the origin host header to be
+	// UseHost causes the origin host header to be
```
Comment (magiconair) [inline route/target.go]:
```
UseHost determines whether the the origin or the target hostname
is used for the upstream request
```

File: route/route.go [hunk 3]
```diff
@@ -73,9 +73,7 @@ func (r *Route) addTarget(service string, targetURL *url.URL, fixedWeight float6
 		t.TLSSkipVerify = r.Opts["tlsskipverify"] == "true"
 		// If the 'usehost' key exists in the map, we set
```
Comment (magiconair) [inline route/route.go]: Can you drop the comment, too please? Don't want to be overly pick but it really is redundant.

Comment (mitchelldavis) [inline route/route.go]: No Worries.  It looks like I need to rerun the build anyways. :-D

**Comment by magiconair** [comment]
I'll let this sit for another day or two before I merge it. `usehost` is concise but also ambiguous. I'll see whether I can come up with something better. Thx for your patience.

**Comment by mitchelldavis** [comment]
@magiconair, how about `usedesthost`?

**Comment by magiconair** [comment]
How about `host=dst`?

**Comment by mitchelldavis** [comment]
ooooo.... that's way better.  Default will be `src`?

**Comment by magiconair** [comment]
Yes. I think that makes more sense and provides some flexibility for the future. boolean flags are always somewhat limiting unless you can be absolutely sure that there will ever only be two choices.

**Comment by magiconair** [comment]
Can you make that change?

**Comment by mitchelldavis** [comment]
I'm on it!

**Comment by magiconair** [comment]
Two small changes and then we can merge it.

**Comment by mitchelldavis** [comment]
Thats a bummer.  It looks like an unstable test. :-(

**Pull request merged**